java/src/org/openqa/selenium/grid/sessionqueue/local/LocalNewSessionQueue.java

```diff
@@ -290,9 +278,15 @@ public boolean retryAddToQueue(SessionRequest request) {
       Lock writeLock = lock.writeLock();
       writeLock.lock();
       try {
+
         if (!requests.containsKey(request.getRequestId())) {
           return false;
-        }
+        } else if (isTimedOut(Instant.now(), requests.get(request.getRequestId()))) {
+          // as we try to re-add a session request that has already expired, force session timeout
+          complete(request.getRequestId(), Either.left(new SessionNotCreatedException("Timed out creating session")));
+          return false;
+        } 
+        
```

Comment:
I don't think this check is needed, the regular timeout handling will pick this up when it is again in the queue.

Reply:
This check is needed because the "timeoutSessions" method only runs every 10 secs (per configuration).
If we remove it, LocalDistributor may try to recreate the session while it's really timed out

Reply:
I think the default value is 1s so and it should be fine*, anyways i don't think this should not be done here.
This is a "atomic" check, when there is 1ms time left before hitting the timeout it will pass and you have to wait for the regular timeout logic to happen.
Just trying to keep the code here as simple as possible as it is already complex, but there might be other opinions. @diemol what do you think about this?

\* Why did you set this timeout so hight in your configuration? 
I am trying to understand / find the root cause for this, to find a good solution to this.

Reply:
The check runs every 10 seconds, so there is a chance of retying a timed-out request.

It does make sense to check if the request has timed out before retrying it. 

However, if you return `true`, `handleNewSessionRequest` in the `LocalDistributor` will invoke `complete` again, which should not cause any issues, but it is just an unneeded call. So maybe return `true` and add a comment about it. 

There is no need for an `else if`, and `if` is enough. Finally, you can reuse the existing `failDueToTimeout` method.

Reply:
ok, done as you suggested